File: src/main/kotlin/dto/file/FileInfo.kt
package org.zak.dto.file

import com.fasterxml.jackson.annotation.JsonFormat
import java.util.*

data class FileInfo(
	val name: String,
	val fullPath: String,
	@JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss")
	val lastModified: Date,
	val size: Long,
	val extension: String,
	val readableSize: String,
	val isDirectory: Boolean = false
)

File: angular_server/src/app/core/model/file-info.ts
import { IconManager } from "../../component/icon-manager/icon-manager";


export class FileInfo{
	constructor(
		public name: string,
		public fullPath: string,
		public lastModified: Date | string,
		public size: bigint,
		public extension: string,
		public readableSize: string,
		public isDirectory: boolean = false
	) {}

	public static getRelativePath(item: FileInfo, path: string): string {
		if (path && item.fullPath.startsWith(path)) {
			return item.fullPath.substring(path.length).replace(/^\//, '');
		}
		return item.fullPath;
	}
}